### PR TITLE
WDPD-116 Reload credit card form after error

### PR DESCRIPTION
**Pull request by ps-mstifter**
### This PR

* reloads the credit card form after an error response was returned from the PaymentGateway
* adds a loading spinner while the credit card form is being loaded
* uses the correct base URL for getting the credit card form

### Jira Links

* https://jira.parkside.at/browse/WDPD-116

### CODE REVIEW

File: out/css/spinner.css [hunk 1]
```diff
@@ -0,0 +1,26 @@
+#cc-spinner {
```
Comment (ps-holler) [inline out/css/spinner.css]: If you make this a class it would be easier to reuse on other pages.

File: out/css/spinner.css [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  display: inline-block;
+  width: 64px;
+  height: 64px;
+}
+
+#cc-spinner:after {
+  content: " ";
```
Comment (ps-holler) [inline out/css/spinner.css]: This can also be empty.
```suggestion
  content: "";
```

File: out/css/spinner.css [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  display: block;
+  width: 46px;
+  height: 46px;
+  margin: 1px;
+  border-radius: 50%;
+  border: 5px solid #009EC0;
+  border-color: #009EC0 transparent #009EC0 transparent;
```
Comment (ps-holler) [inline out/css/spinner.css]: If you use `border-style` instead, you don't have to repeat the color (so it's easier to maintain).
```suggestion
  border-style: solid none;
```

File: out/css/spinner.css [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  animation: cc-spinner 1.2s linear infinite;
+}
+
+@keyframes cc-spinner {
```
Comment (ps-holler) [inline out/css/spinner.css]: Consider adding at least `@-webkit-keyframes` for a wider browser support.

File: out/js/credit_card_form.js [hunk 5]
```diff
@@ -6,22 +6,28 @@
  * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
  */
 /* global WirecardPaymentPage */
-var ModuleCreditCardForm = (function ($) {
-
+var ModuleCreditCardForm = (function($) {
```
Comment (ps-holler) [inline out/js/credit_card_form.js]:
```suggestion
var ModuleCreditCardForm = (function ($) {
```
Please don't remove spaces from anonymous functions (same applies for the lines below).

File: out/js/credit_card_form.js [hunk 6]
```diff
@@ -32,50 +38,100 @@ var ModuleCreditCardForm = (function ($) {
...
+    var parsedObj = null;
+
+    try {
+      var dataObj = JSON.parse(responseString);
+
+      var requestDataObj = JSON.parse(dataObj["requestData"]);
+
+      parsedObj = requestDataObj;
+    } catch (ex) {
+      // eslint-disable-next-line no-console
```
Comment (ps-holler) [inline out/js/credit_card_form.js]: a) Why log the error to the console?
b) Since when are we using ESLint?

File: out/css/spinner.css (hunk 2 above)
Comment (tomazpu) [reply out/css/spinner.css]: please make it empty and not a whitespace

File: out/js/credit_card_form.js [hunk 7]
```diff
@@ -6,22 +6,28 @@
  * https://github.com/wirecard/oxid-ee/blob/master/LICENSE
  */
 /* global WirecardPaymentPage */
```
Comment (tomazpu) [inline out/js/credit_card_form.js]: Please update the name

File: out/js/credit_card_form.js (hunk 5 above)
Comment (ps-amehic) [reply out/js/credit_card_form.js]: Codacy showed an error when the space was there.

File: out/js/credit_card_form.js [hunk 8]
```diff
@@ -6,22 +6,28 @@
...
+var ModuleCreditCardForm = (function($) {
   var debug = false;
 
+  var requestData = null;
+
   function getOrderButton() {
     return $("#orderConfirmAgbBottom button[type = 'submit']");
   }
 
   function callback(response) {
```
Comment (tomazpu) [inline out/js/credit_card_form.js]: am.. where do you use response?

File: Extend/Controller/OrderController.php [hunk 9]
```diff
@@ -271,28 +271,85 @@ private function _handleTransaction($oBasket, $oOrder)
...
+         * @var $oBasket Basket
+         */
+        $oBasket = $this->getBasket();
+        $oTransaction = $this->createCreditCardTransactionFromBasket($oBasket);
+
+        $oPayment = PaymentMethodHelper::getPaymentById($oBasket->getPaymentId());
+        $sPaymentAction = $this->_getPaymentAction($oPayment->oxpayments__wdoxidee_transactionaction->value);
+        $sLanguageCode = Registry::getLang()->getLanguageAbbr();
+
+        $oTransactionService = $this->_getTransactionService();
```
[the remark this answers is not in the document]
Comment (ps-amehic) [reply Extend/Controller/OrderController.php]: Method _getCreditCardPaymentMethodConfig is getting the config with headers. See getConfig in PaymentMethod.php.

File: Extend/Controller/OrderController.php [hunk 10]
```diff
@@ -271,28 +271,85 @@ private function _handleTransaction($oBasket, $oOrder)
...
+     *
+     * @since 1.0.0
+     */
+    private function _getCreditCardFormRequestData()
+    {
+        /**
+         * @var $oBasket Basket
+         */
+        $oBasket = $this->getBasket();
+        $oTransaction = $this->createCreditCardTransactionFromBasket($oBasket);
```
[the remark this answers is not in the document]
Comment (ps-amehic) [reply Extend/Controller/OrderController.php]: Yes. PaymentGateway.php has a method createTransaction which is being used to get all the additional data.

File: out/css/spinner.css [hunk 11]
```diff
@@ -0,0 +1,35 @@
...
+    transform: rotate(0deg);
+  }
+  100% {
+    transform: rotate(360deg);
+  }
+}
+
+@-webkit-keyframes cc-spinner {
+  0% {
+    transform: rotate(0deg);
```
Comment (ps-holler) [inline out/css/spinner.css]:
```suggestion
    -webkit-transform: rotate(0deg);
```

**Pull request merged**